bids/layout/layout.py

```diff
@@ -293,7 +293,10 @@ def _sanitize_query_dtypes(self, entities):
                 continue
             try:
                 if isinstance(val, (list, tuple)):
-                    entities[name] = [ents[name]._astype(v) for v in val]
+                    entities[name] = [
+                        v if isinstance(v, enum.Enum) else ents[name]._astype(v)
```

Comment:
I see we're using `Enum` in L292 for some reason, but I think we want to be more specific.

```suggestion
                        v if isinstance(v, Query) else ents[name]._astype(v)
```